src/components/LiveCursors/use-spaces.ts

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react'
+import Spaces, { Space } from '@ably-labs/spaces'
+import { assertConfiguration } from '@ably-labs/react-hooks'
+import { useEffect } from 'react'
+interface UserColors {
+  gradientStartClass: string
+  gradientEndClass: string
+  cursorStartColor: string
+  cursorEndColor: string
+}
+
+// 💡 This hook is used to create a new space and enter it
+const useSpaces = (name: string, userName: string, userColors: UserColors) => {
```

Comment:
I agree, but was thinking it might be easier from a cognitive load perspective for someone interested in just one or the other to have them separated out? wdyt?

Reply:
these eventually will be replaced by a generic hook anyway - it's supposed to just be a way to set up spaces (not cursors or anything else) so I think we should make the effort to make it so.